### PR TITLE
feat: OSS application metrics

### PR DESCRIPTION
**Pull request by jsgoudar**
#### What this PR does / why we need it:
PR add business metrics to the retail application and shows the metrics collected on grafana
#### Which issue(s) this PR fixes:
new feature - business metrics
Fixes #

#### Quality checks

- [y ] My content adheres to the style guidelines
- [y ] I ran `make test` or `make e2e-test` and it was successful

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by niallthomson** [comment]
We're going to need to update the retail sample images used from `latest` to `0.3.0` before this can be merged.

**Comment by jsgoudar** [comment]
> We're going to need to update the retail sample images used from `latest` to `0.3.0` before this can be merged.

images updated

**Review by niallthomson — CHANGES_REQUESTED** [review]

File: website/docs/observability/open-source-metrics/business-metrics.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+Access the UI in your web browser ,following instructions detailed in workshop under [Exposing applications](https://www.eksworkshop.com/docs/fundamentals/exposing/). Once you load the ui page, add a couple of watch orders so we can pull the metrics.
+
+The application code will capture the business metrics outline above. These metrics are exported to Amazon Managed Prometheus. We will use Amazon Managed Grafana to visualize the metrics. 
+
+Open the retail sample application and place a few orders using different quantities of the watches.
+
+Open the Grafana dashboard![Grafana dashboard](./assets/business-metrics-dashboard.png)
+
+Go to the dashboard section and click on the dashboard **Retail Business Metrics** and let's review the panels wintin the dashboard.
```
[the remark this answers is not in the document]
Comment (jsgoudar) [reply website/docs/observability/open-source-metrics/business-metrics.md]: Incorporate the review comments in the commit

**Comment by niallthomson** [comment]
Here is how you can run the load generator to create orders:

```cat <<'EOF' | kubectl apply -f -
apiVersion: v1
kind: Pod
metadata:
  name: load-generator
spec:
  containers:
  - name: artillery
    image: artilleryio/artillery:2.0.0-31
    args:
    - "run"
    - "-t"
    - "http://ui.default.svc"
    - "/scripts/scenario.yml"
    volumeMounts:
    - name: scripts
      mountPath: /scripts
  initContainers:
  - name: setup
    image: public.ecr.aws/aws-containers/retail-store-sample-utils:load-gen.0.3.0
    command:
    - bash
    args:
    - -c
    - "cp /artillery/* /scripts"
    volumeMounts:
    - name: scripts
      mountPath: "/scripts"
  volumes:
  - name: scripts
    emptyDir: {}
EOF
```

**Comment by normalfaults** [comment]
Discussed with @svennam92 : @jsgoudar can you change the language from business metric to application metric ?

File: terraform/modules/cluster/addons.tf [hunk 2]
```diff
@@ -526,6 +526,21 @@ module "eks_blueprints_kubernetes_grafana_addon" {
...
+    name      = "business-metrics-dashboard"
+    namespace = "grafana"
+
+    labels = {
+      grafana_dashboard = 1
+    }
+  }
+  
+  data = {
+    "business-metrics-dashboard.json" = file("${path.module}/templates/dashboards/business-metrics-dashboard.json")
```
Comment (niallthomson) [inline terraform/modules/cluster/addons.tf:540]: Rename the JSON file as above

File: terraform/modules/cluster/templates/grafana.yaml [hunk 3]
```diff
@@ -48,18 +48,22 @@ dashboardProviders:
...
+      orgId: 1
+      folder: "retail-app-metrics"
+      type: file
+      disableDeletion: false
+      editable: false
+      options:
+        path: /var/lib/grafana/dashboards/bmetrics       
+
+dashboardsConfigMaps:
+  bmetrics: "business-metrics-dashboard"
```
Comment (niallthomson) [inline terraform/modules/cluster/templates/grafana.yaml:62]: Rename to 'order-service-metrics-dashboard'

File: website/docs/observability/open-source-metrics/application-metrics.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
+---
+title: "Custom Metrics"
```
Comment (niallthomson) [inline website/docs/observability/open-source-metrics/application-metrics.md:2]: Rename to "Application Metrics"

File: website/docs/observability/open-source-metrics/application-metrics.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+sidebar_position: 50
+---
+
+Custom metrics are user defined metrics that help track your application KPIs. These metrics help analyze data and derive insights to meet your application needs. Let's look at some custom metrics using AWS Distro for OpenTelemetry and visualize the metrics using grafana.
```
Comment (niallthomson) [inline website/docs/observability/open-source-metrics/application-metrics.md:6]: "Application metrics are user-defined metrics"

File: website/docs/observability/open-source-metrics/application-metrics.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+The custom metrics we want to capture and show in the dashboard for the sample application are:
+- number of total orders created
+- most popular watch type order, based on number of each watch type ordered
+- total cost of each order
+
+The "retail-store-sample-app" has already been instrumented to capture the above mentioned metrics. The OrderMetrics class uses Spring Boot [Micrometer](https://spring.io/blog/2018/03/16/micrometer-spring-boot-2-s-new-application-metrics-collector) a metrics collection facade which collects metrics data from the application with a vendor neutral API. 
+
+Access the ui in your web browser ,following instructions detailed in workshop under [Exposing applications](https://www.eksworkshop.com/docs/fundamentals/exposing/). Once you load the ui page, add a couple of watch orders so we can generate the metrics.
```
Comment (niallthomson) [inline website/docs/observability/open-source-metrics/application-metrics.md:15]: You should not refer to other labs. Remove manually loading the application from this section, just use the load generator.

File: website/docs/observability/open-source-metrics/application-metrics.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+This panel displays the toal number of orders placed by the user in the given time range.
+![Total Orders Placed](./assets/totalOrders.png)
+
+This panel displays the number of each watch type ordered, displayed as a pie chart.
+![Most Popular Watch Ordered](./assets/watchCount.png)
+
+This panel displays the toal price of the order in the given time range.
+![Total Order Price](./assets/totalOrderPrice.png)
+
+Once you're satisfied with observing metrics, you can end the load generator with `Ctrl+C` and stop the load generator using the below command.
```
Comment (niallthomson) [inline website/docs/observability/open-source-metrics/application-metrics.md:73]: You can remove the Ctrl+C comment there, deleting the pod is enough

File: website/docs/observability/open-source-metrics/application-metrics.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+The application code will capture the application metrics outline above. These metrics are exported to Amazon Managed Prometheus. We will use Amazon Managed Grafana to visualize the metrics. 
+
+Use the below script to run load-generator to place watch orders to capture the custom metrics.
+
+```kubectl wait --for=condition=Ready --timeout=180s pods \
+cat <<'EOF' | kubectl apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: load-generator
```
Comment (niallthomson) [inline website/docs/observability/open-source-metrics/application-metrics.md:26]: Please run this in the namespace `other`

**Review by niallthomson — CHANGES_REQUESTED** [review]
Thanks for the PR @jsgoudar, great section to add. Personally I'd like to see more explanation about how this mechanism works. For example I'm less interested in how the order service implements metrics and more interested that the each sample application component exposes a prometheus endpoint as a consistent metrics implementation. This can be demonstrated with `curl` commands to show where these application metrics are actually being picked up from.

We should also provide some explanation with regards to how the OpenTelemetry collector is configured to scrape these metrics. Otherwise the user is left to figure out by themselves how the metrics get to Prometheus and then Grafana.

You could also provide a brief explanation of one of the PromQL queries used to construct the dashboard. This would help the reader trace the metrics from the application through to the dashboard.

Couple of other minor things:
1. The sample application has been updated to 0.4.0 so you can remove those file changes
2. It seems to have picked up some changes to the Cloud9 access page, please revert that file

File: terraform/modules/cluster/addons.tf (hunk 2 above)
Comment (jsgoudar) [reply terraform/modules/cluster/addons.tf]: done

File: website/docs/observability/open-source-metrics/application-metrics.md (hunk 4 above)
Comment (jsgoudar) [reply website/docs/observability/open-source-metrics/application-metrics.md]: done

File: website/docs/observability/open-source-metrics/application-metrics.md (hunk 6 above)
Comment (jsgoudar) [reply website/docs/observability/open-source-metrics/application-metrics.md]: done

File: website/docs/observability/open-source-metrics/application-metrics.md (hunk 8 above)
Comment (jsgoudar) [reply website/docs/observability/open-source-metrics/application-metrics.md:25]: done

**Comment by normalfaults** [comment]
LGTM @niallthomson - ready to merge

**Pull request merged**